### PR TITLE
Draft of 0.4.0 release notes

### PR DESCRIPTION
**Pull request by travisbrown**
I'll update if any last-minute stuff makes it in before the release. Rendered version available [here](https://github.com/travisbrown/cats/blob/topic/0.4.0-release-notes/CHANGES.md) for now.

### CODE REVIEW

**Comment by fthomas** [comment]
:+1: That's an impressively long list of changes!

**Comment by non** [comment]
Thanks @travisbrown!

**Comment by non** [comment]
Cats 0.4.0 is on Sonatype. Let's wait until it hits Maven Central and then merge this PR. :+1:

**Comment by non** [comment]
Actually we can merge this now but wait to update the README, etc until the sync happens.

**Comment by DavidGregory084** [comment]
@non @travisbrown Just thought I'd point out that #809 went into topic/kernel so I don't think it will be in 0.4.0

**Comment by travisbrown** [comment]
@DavidGregory084 Thanks for catching that—not sure how it slipped through…

**Comment by DavidGregory084** [comment]
No problem, just spotted it while I was reading the :metal: epic list of changes :metal:

File: CHANGES.md [hunk 1]
```diff
@@ -1,3 +1,254 @@
...
+* [#701](https://github.com/typelevel/cats/pull/701): Better documentation for
+  the Travis CI script
+* [#787](https://github.com/typelevel/cats/pull/787): Support for cross-module
+  Scaladoc links
+
+Known issues:
+
+* [#702](https://github.com/typelevel/cats/pull/702): This change identified and
+  fixed a stack safety bug in `foldMap` on `Free`, but raised other issues (see
+  [#712](https://github.com/typelevel/cats/issues/712)) and was reverted in
```
Comment (refried) [inline CHANGES.md]: Extra ‘)’?

Comment (travisbrown) [inline CHANGES.md]: @refried What do you mean? The first closes the link and the second the parenthetical.

**Pull request merged**